File: H5/H5/Resources/.generated/h5/Ref.js
    H5.define("H5.Ref$1", function (T) { return {
        statics: {
            methods: {
                op_Implicit: function (reference) {
                    return reference.Value;
                }
            }
        },
        fields: {
            getter: null,
            setter: null
        },
        props: {
            Value: {
                get: function () {
                    return this.getter();
                },
                set: function (value) {
                    this.setter(value);
                }
            },
            v: {
                get: function () {
                    return this.Value;
                },
                set: function (value) {
                    this.Value = value;
                }
            }
        },
        ctors: {
            ctor: function (getter, setter) {
                this.$initialize();
                this.getter = getter;
                this.setter = setter;
            }
        },
        methods: {
            toString: function () {
                return H5.toString(this.Value);
            },
            valueOf: function () {
                return this.Value;
            }
        }
    }; });

File: H5/H5/H5/Ref.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace H5
{
    public class Ref<T>
    {
        private Func<T> getter;
        private Action<T> setter;

        public T Value
        {
            get
            {
                return getter();
            }
            set
            {
                setter(value);
            }
        }

        private T v
        {
            get
            {
                return Value;
            }
            set
            {
                Value = value;
            }
        }

        public Ref(Func<T> getter, Action<T> setter)
        {
            this.getter = getter;
            this.setter = setter;
        }

        public static implicit operator T(Ref<T> reference)
        {
            return reference.Value;
        }

        public override string ToString()
        {
            return Value.ToString();
        }

        [Name("valueOf")]
        public override object ValueOf()
        {
            return Value;
        }
    }
}
